indexd/index/drivers/alchemy.py

```diff
@@ -658,47 +678,51 @@ def update(self, did, rev, changing_fields):
-                if key == 'urls':
-                    for url in record.urls:
-                        session.delete(url)
-
-                    record.urls = [
-                        IndexRecordUrl(did=record.did, url=url)
-                        for url in value
-                    ]
-
-                elif key == 'acl':
-                    for ace in record.acl:
-                        session.delete(ace)
-
-                    record.acl = [
-                        IndexRecordACE(did=record.did,ace=ace)
-                        for ace in value
-                    ]
-
-                elif key == 'metadata':
-                    for md_record in record.index_metadata:
-                        session.delete(md_record)
-
-                    record.index_metadata = [
-                        IndexRecordMetadata(
-                            did=record.did,
-                            key=m_key,
-                            value=m_value
-                        )
-                        for m_key, m_value in value.items()]
+            if 'urls' in changing_fields:
+                for url in record.urls:
+                    session.delete(url)
+
+                record.urls = [
+                    IndexRecordUrl(did=record.did, url=url)
+                    for url in changing_fields['urls']
+                ]
+
+            if 'acl' in changing_fields:
+                for ace in record.acl:
+                    session.delete(ace)
+
+                record.acl = [
+                    IndexRecordACE(did=record.did, ace=ace)
+                    for ace in changing_fields['acl']
+                ]
+
+            if 'metadata' in changing_fields:
+                for md_record in record.index_metadata:
+                    session.delete(md_record)
+
+                record.index_metadata = [
+                    IndexRecordMetadata(
+                        did=record.did,
+                        key=m_key,
+                        value=m_value
+                    )
+                    for m_key, m_value in changing_fields['metadata'].items()]
 
-                elif key == 'urls_metadata':
-                    for url in record.urls:
-                        for url_metadata in url.url_metadata:
-                            session.delete(url_metadata)
+            if 'urls_metadata' in changing_fields:
+                for url in record.urls:
+                    for url_metadata in url.url_metadata:
+                        session.delete(url_metadata)
 
-                    create_urls_metadata(value, record, session)
+                create_urls_metadata(
+                    changing_fields['urls_metadata'],
+                    record,
+                    session,
+                )
 
-                # No special logic needed for other updates.
-                # ie file_name, version, etc
-                else:
+            for key, value in changing_fields.items():
+                if key not in special_fields:
```

Comment:
'special_fields' sounds confusing, can you rename to composite_fields? and you need to add a comment to say urls_metadata need to be processed after urls